### PR TITLE
[docs] Fix docs links and pagination sentence

### PR DESCRIPTION
**Pull request by ZeeshanTamboli**
Just a few small fixes in docs.

Preview: 
https://deploy-preview-2381--material-ui-x.netlify.app/components/data-grid/components
https://deploy-preview-2381--material-ui-x.netlify.app/components/data-grid/editing
https://deploy-preview-2381--material-ui-x.netlify.app/components/data-grid/pagination/#server-side-pagination

### CODE REVIEW

File: docs/src/pages/components/data-grid/editing/editing.md [hunk 1]
```diff
@@ -75,13 +75,13 @@ To validate the value in the cells, use `onEditRowsModelChange` to set the `erro
 If this attribute is true, the value will never be commited.
 This prop is invoked when a change is triggered by the edit cell component.
 
-Alternatively, you can use the `GridEditRowsModel` state mentioned in the [Control editing](#control-editing) section.
+Alternatively, you can use the `GridEditRowsModel` state mentioned in the [Control editing](#controlled-editing) section.
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/editing/editing.md:78]:
```suggestion
Alternatively, you can use the `GridEditRowsModel` state mentioned in the [Controlled editing](#controlled-editing) section.
```

File: docs/src/pages/components/data-grid/components/components.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ title: Data Grid - Components
 ## Overriding components
 
 As part of the customization API, the grid allows you to override internal components with the `components` prop.
-The prop accepts an object of type [`GridSlotsComponent`](/api/data-grid/#slots).
+The prop accepts an object of type [`GridSlotsComponent`](/api/data-grid/data-grid/#slots).
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/components/components.md:13]: **Off-topic**

We have talked about using these URLs in v5
```suggestion
The prop accepts an object of type [`GridSlotsComponent`](/api/x-data-grid/data-grid/#slots).
```

Using the name of the package then the name of the module exported: https://www.notion.so/mui-org/Sitemap-and-URL-structure-rearrange-ca604f628c17438db2d987961fcb8ffe

File: docs/src/pages/components/data-grid/components/components.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ title: Data Grid - Components
...
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/components/components.md:12]: **Off-topic**

We have talked about using these URLs in v5
```suggestion
The prop accepts an object of type [`GridSlotsComponent`](/api/x-data-grid/data-grid/#slots).
```

Using the name of the package then the name of the module exported: https://www.notion.so/mui-org/Sitemap-and-URL-structure-rearrange-ca604f628c17438db2d987961fcb8ffe

Comment (ZeeshanTamboli) [reply docs/src/pages/components/data-grid/components/components.md:12]: Cool. I cannot access MUI notion. Is it possible to give me access?

Comment (m4theushw) [reply docs/src/pages/components/data-grid/components/components.md:12]: I've added another item to #2313 to keep track. We will need redirect rules to not send users to a 404. This is the proposed structure:

- [mui.com/components/x-data-grid](http://mui.com/components/x-data-grid) → X component main page.
    - [mui.com/api/x-data-grid/grid-toolbar](http://mui.com/api/x-data-grid/grid-toolbar) → Exposing smaller components with Data Grid, for example.
    - [mui.com/api/x-data-grid-pro/grid-toolbar](http://mui.com/api/x-data-grid/grid-toolbar) → Similarly, we can do it for those smaller components on the paid version.

**Review by m4theushw — APPROVED** [review]

**Pull request merged**